src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String newLine = System.lineSeparator();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromFormatter = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToFormatter = LocalDate.parse(dateTo, formatter).plusDays(1);
+        int[] salaries = new int[names.length];
+        for (int i = 0; i < names.length; i++) {
+            for (String dateFinding: data) {
+                String[] dateSplit = dateFinding.split(" ");
+                LocalDate dateWork = LocalDate.parse(dateSplit[0], formatter);
+                if (names[i].equals(dateSplit[1]) && dateWork.isAfter(dateFromFormatter)
+                        && dateWork.isBefore(dateToFormatter)) {
+                    salaries[i] += Integer.parseInt(dateSplit[2]) * Integer.parseInt(dateSplit[3]);
```

Comment:
Please introduce constants for magic numbers/strings
```suggestion
                String[] dateSplit = dateFinding.split(SPACE_SEPARATOR);
                LocalDate dateWork = LocalDate.parse(dateSplit[DATE_INDEX], formatter);
                if (names[i].equals(dateSplit[NAME_INDEX]) && dateWork.isAfter(dateFromFormatter)
                        && dateWork.isBefore(dateToFormatter)) {
                    salaries[i] += Integer.parseInt(dateSplit[WORKED_HOURS_INDEX]) * Integer.parseInt(dateSplit[PER_HOUR_SALARY_INDEX]);
```